extensions/ql-vscode/src/query-history.ts

```diff
@@ -603,6 +604,7 @@ export class QueryHistoryManager extends DisposableObject {
         t: 'variant-analysis',
         status: QueryStatus.InProgress,
         completed: false,
+        queryId: nanoid(),
```

Comment:
Does this look like the right place to create the `queryId`? (as in, in the `onVariantAnalysisAdded` event)

Reply:
Yep!

Reply:
Creating the ID here looks fine to me too.

The remote queries version creates the ID one step and includes it in the `NewQueryEvent`, but I don't think this is necessarily better and I don't think it makes a difference.

It's also something we can easily change in the future so it's very low risk.